gui/integrity-dialog.cpp

```diff
@@ -0,0 +1,425 @@
+
+	Common::JSONArray filesObject;
+
+	for (Common::StringArray fileChecksum : fileChecksums) {
+		Common::JSONObject file;
+		Common::String relativePath = fileChecksum[0].substr(gamePath.size() + 1); // +1 to remove the first '/'
+		file.setVal("name", new Common::JSONValue(relativePath));
+
+		auto tempNode = Common::FSNode(fileChecksum[0]);
+		Common::File tempFile;
+		if (!tempFile.open(tempNode))
+			continue;
+		long long fileSize = tempFile.size();
+		tempFile.close();
+
+		file.setVal("size", new Common::JSONValue(fileSize));
+
+		Common::JSONArray checksums;
+		Common::StringArray checkcodes = {"md5", "md5-5000", "md5-1M", "md5-t-5000"};
+
+		int index = -1;
+		for (Common::String val : fileChecksum) {
+			index++;
+
+			Common::JSONObject checksum;
+			if (index < 1) {
+				continue;
+			}
+
+			checksum.setVal("type", new Common::JSONValue(checkcodes[index - 1]));
+			checksum.setVal("checksum", new Common::JSONValue(val));
+
+			checksums.push_back(new Common::JSONValue(checksum));
+		}
+		file.setVal("checksums", new Common::JSONValue(checksums));
+
+		filesObject.push_back(new Common::JSONValue(file));
+	}
+
+	requestObject.setVal("files", new Common::JSONValue(filesObject));
+
+	Common::JSONValue *request = new Common::JSONValue(requestObject);
+	return request;
+}
+
+void IntegrityDialog::checksumResponseCallback(Common::JSONValue *r) {
+	debug(r->stringify().c_str());
+	Common::String messageText = IntegrityDialog::parseJSON(r);
+	MessageDialog result(messageText);
+	result.reflowLayout();
+}
+
+void IntegrityDialog::errorCallback(Networking::ErrorResponse error) {
+	debug("ERROR %ld: %s", error.httpResponseCode, error.response.c_str());
+}
+
+void IntegrityDialog::sendJSON() {
+	auto conn = new Networking::PostRequest(g_state->endpoint,
+											new Common::Callback<IntegrityDialog, Common::JSONValue *>(this, &IntegrityDialog::checksumResponseCallback),
+											new Common::Callback<IntegrityDialog, Networking::ErrorResponse>(this, &IntegrityDialog::errorCallback));
+
+	Common::JSONValue *json = generateJSONRequest(
+		g_state->gamePath, g_state->gameid, g_state->engineid, g_state->extra, g_state->platform, g_state->language);
+	conn->setJSONData(json);
+	conn->setContentType("application/json");
+	conn->start();
+	delete json;
+}
+
+Common::String IntegrityDialog::parseJSON(Common::JSONValue *response) {
+	Common::String messageText;
+
+	Common::JSONObject responseObject = response->asObject();
+	int responeError = responseObject.getVal("error")->asIntegerNumber();
+
+	if (responeError == -1) { // Unknown variant
+		long long fileset = responseObject.getVal("fileset")->asIntegerNumber();
+		messageText =
+			Common::String::format("Your set of game files seems to be unknown to us. If you are sure that this is a valid unknown variant, please send the following e-mail to integrity@scummvm.org \n\
+		The game of fileset %lld seems to be an unknown game variant. \n\
+		The details of the game : %s, %s, %s, %s, %s",
```

Comment:
Where is the part asking to send an e-mail and copy all of this info to the clipboard?